massa-event-cache/src/ser_deser.rs

```diff
@@ -0,0 +1,294 @@
+            block_id_ser: OptionSerializer::new(BlockIdSerializer::new()),
+            op_id_ser: OptionSerializer::new(OperationIdSerializer::new()),
+            data_ser: StringSerializer::new(U64VarIntSerializer::new()),
+        }
+    }
+}
+
+impl Default for SCOutputEventSerializer {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl Serializer<SCOutputEvent> for SCOutputEventSerializer {
+    fn serialize(&self, value: &SCOutputEvent, buffer: &mut Vec<u8>) -> Result<(), SerializeError> {
+        // context
+        self.slot_ser.serialize(&value.context.slot, buffer)?;
+        self.block_id_ser.serialize(&value.context.block, buffer)?;
+        buffer.push(u8::from(value.context.read_only));
+        self.u64_ser
+            .serialize(&value.context.index_in_slot, buffer)?;
+        // Components
+        let call_stack_len_ = value.context.call_stack.len();
+        let call_stack_len = u32::try_from(call_stack_len_).map_err(|_| {
+            SerializeError::GeneralError(format!(
+                "Cannot convert component_len ({}) to u32",
+                call_stack_len_
+            ))
+        })?;
+        // ser vec len
+        self.u32_ser.serialize(&call_stack_len, buffer)?;
+        for address in value.context.call_stack.iter() {
+            self.addr_ser.serialize(address, buffer)?;
+        }
+        self.op_id_ser
+            .serialize(&value.context.origin_operation_id, buffer)?;
+        buffer.push(u8::from(value.context.is_final));
+        buffer.push(u8::from(value.context.is_error));
+
+        // data
+        self.data_ser.serialize(&value.data, buffer)?;
+        Ok(())
+    }
+}
+
+/// SCOutputEvent deserializer
+pub struct SCOutputEventDeserializer {
+    u64_deser: U64VarIntDeserializer,
+    u32_deser: U32VarIntDeserializer,
+    slot_deser: SlotDeserializer,
+    addr_deser: AddressDeserializer,
+    block_id_deser: OptionDeserializer<BlockId, BlockIdDeserializer>,
+    op_id_deser: OptionDeserializer<OperationId, OperationIdDeserializer>,
+    data_deser: StringDeserializer<U64VarIntDeserializer, u64>,
+}
+
+impl SCOutputEventDeserializer {
+    pub fn new(args: SCOutputEventDeserializerArgs) -> Self {
+        Self {
+            u64_deser: U64VarIntDeserializer::new(Included(0), Included(u64::MAX)),
+            // FIXME: is there some limit for call stack len?
+            u32_deser: U32VarIntDeserializer::new(Included(0), Included(u32::MAX)),
+            slot_deser: SlotDeserializer::new(
+                (Included(0), Included(u64::MAX)),
+                (Included(0), Excluded(args.thread_count)),
+            ),
+            addr_deser: Default::default(),
+            block_id_deser: OptionDeserializer::new(BlockIdDeserializer::new()),
+            op_id_deser: OptionDeserializer::new(OperationIdDeserializer::new()),
+            data_deser: StringDeserializer::new(U64VarIntDeserializer::new(
+                Included(0),
+                Included(u64::MAX),
```

Comment:
note: this should be safe if we are only talking about the cache. But if in the future we use this deserializer for other purposes, this will induce a security threat. It might be worth to introduce limits here as a good practice